src/components/trackInfo/trackInfo.js

```diff
@@ -26,12 +27,18 @@ export class TrackInfo extends React.Component {
     componentDidMount() {
         window.scrollTo(0, 0);
 
-        const { tracks } = this.props;
+        const { tracks, loadTracks, updateCurrentTrack } = this.props;
         const trackId = this.getTrackId();
         const track = this.getTrackById(trackId);
 
         if (!track || !tracks.length) {
-            this.props.loadTracks(trackId);
+            const promise = loadTracks(trackId);
+            // update info about current track after tracks has loaded
+            promise
+                .then(
+                    () => updateCurrentTrack(trackId),
+                    (error) => { throw new Error(error); },
```

Comment:
Why not promise.catch?

Reply:
I don't know. It works :) 
But with promise.catch code looks more cleaner.